java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -283,33 +281,26 @@ public void shouldGetFilesThatAreReadyForGC() throws Exception {
         //      just been marked as ready for GC
         FileInfo fileInfo3 = FileInfo.wholeFile()
                 .filename("file3")
-                .fileStatus(FileInfo.FileStatus.READY_FOR_GARBAGE_COLLECTION)
+                .fileStatus(FileInfo.FileStatus.ACTIVE)
                 .partitionId(partition.getId())
                 .numberOfRecords(100L)
                 .build();
-        stateStore.fixTime(file3Time);
         stateStore.addFile(fileInfo3);
+        stateStore.fixTime(file1Time);
+        stateStore.atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFiles(List.of(fileInfo1), List.of());
+        stateStore.fixTime(file3Time);
+        stateStore.atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFiles(List.of(fileInfo3), List.of());
 
         // When 1
         stateStore.fixTime(file1GCTime);
-        Iterator<FileInfo> readyForGCFilesIterator = stateStore.getReadyForGCFiles();
-
         // Then 1
-        assertThat(stateStore.getReadyForGCFilenamesBefore(file2Time)).containsExactly("file1");
-        assertThat(readyForGCFilesIterator).toIterable()
-                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
-                .containsExactly(fileInfo1);
-
+        assertThat(stateStore.getReadyForGCFilenamesBefore(file2Time))
+                .containsExactly("file1");
         // When 2
         stateStore.fixTime(file3GCTime);
-        readyForGCFilesIterator = stateStore.getReadyForGCFiles();
-
         // Then 2
         assertThat(stateStore.getReadyForGCFilenamesBefore(file3Time.plus(Duration.ofMinutes(1))))
                 .containsExactlyInAnyOrder("file1", "file3");
-        assertThat(readyForGCFilesIterator).toIterable()
-                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
-                .containsExactlyInAnyOrder(fileInfo1, fileInfo3);
```

Comment:
The fixTime calls here aren't needed now that the time filter is passed directly in getReadyForGCFilenamesBefore.